apps/wakunode2/wakunode2_validator_signed.nim

```diff
@@ -49,23 +51,53 @@ proc withinTimeWindow*(msg: WakuMessage): bool =
     return true
   return false
 
-proc addSignedTopicValidator*(w: WakuRelay, topic: PubsubTopic, publicTopicKey: SkPublicKey) =
-  debug "adding validator to signed topic", topic=topic, publicTopicKey=publicTopicKey
+proc getKeyFromTopic*(protectedTopic: PubsubTopic): Result[string, string] =
+  # example: /waku/2/signed:0x2ea1f2ec2da14e0e3118e6c5bdfa0631785c76cd/proto
+  # see https://rfc.vac.dev/spec/57/#dos-protection
+  let topicTag = "signed"
+  let parts = protectedTopic.split("/")
+  if parts.len != 5:
```

Comment:
I've added some comments on the [RFC PR](https://github.com/vacp2p/rfc/pull/601) that would make this an invalid check. For one, we've deprecated the use of the superfluous `/proto` on pubsub topic names. We also want people to still be able to define their own topic names for disambiguation too (what if multiple topics need to be signed with the same key?).  We also want to be able to define the static shard in the topic name for Status Communities.